app/models/spree/order_decorator.rb

```diff
@@ -289,7 +289,8 @@ def associate_customer
 
   def ensure_customer
     unless associate_customer
-      self.customer = Customer.create(enterprise: distributor, email: email_for_customer, user: user)
+      customer_name = bill_address.andand.full_name
+      self.customer = Customer.create(enterprise: distributor, email: email_for_customer, user: user, name: customer_name, bill_address: bill_address.andand.clone, ship_address: ship_address.andand.clone)
```

Comment:
I think that it would be better to set `bill_address` and `ship_address`, and therefore `name` in the `set_default_bill_address` and `set_default_ship_address` methods on `app/controllers/checkout_controller.rb`. Otherwise, the addresses could be set to nil here, because they may not exist yet when ensure_customer is called. If we do it in the controller instead, we always know that we have address to assign to the customer.  

What do you think?